### ISSUE TITLE
L.Control.Layer throws error if <input> inside layer label

### ISSUE THREAD

**Issue by IvanSanchez** [Leaflet/Leaflet#5116]
Small non-critical bug on an edge case, courtesy of @tomchadwin from [a gis.stackexchange question](https://gis.stackexchange.com/questions/218286/adding-input-to-leaflet-1-0-0-layers-control-throws-an-error/218290).

If a `L.Control.Layers` has an `<input>` inside the label for a layer, like...

```js
    var layersControl = L.control.layers({
      "<input id='positron-slider' type='range' min='0' max='1' step='0.1'></input> Positron": positron,
      "<input id='darkmatter-slider' type='range' min='0' max='1' step='0.1'></input> Dark Matter": darkmatter,
    }).addTo(map);
```

(Full playground at https://playground-leaflet.rhcloud.com/wipo/edit?html,output )

Then the logic inside the `_checkDisabledLayers` and `_onInputClick` will try to iterate through that input as well, and throw an error during the process.

The problem lies with the way that the code iterates through the checkboxes and radiobuttons. There is an in-depth explanation in the gis.stackexchange link above:

```js
		var inputs = this._form.getElementsByTagName('input'),
```

The idea is to replace that `getElementsByTagName()` query with a new private property like `this._inputs`. That should be a plain array, populated in the `_addItem` private method, with something like `this._inputs.push(input);`. Care should be taken when calling `removeLayer`.

This sounds fairly easy and non-critical, so I'm going to let this up for grabs (ping @yourfirstpr ). Anybody wanting to work on this should:

- Say something in the issue comments
- Fork the repo
- Clone the fork
- Copy-paste some file from the `debug/test` directory, and put the code from my playground in there. That's the easiest way to debug this.
- Make the necessary modifications to [`/src/control/Control.Layers.js`](https://github.com/Leaflet/Leaflet/blob/master/src/control/Control.Layers.js)
- Test, test, test
- For bonus points, add a unit test to `/spec/suites/Control.LayersSpec`
- Commit the changes and push to your fork
- Create a pull request

**Comment by IvanSanchez** [comment]
@serdarkacka @tomchadwin Anybody wants to take this one? Should be easy enough.

**Comment by tomchadwin** [comment]
I'll take a look, but I'm not confident to be able to fix it myself, so by all means others can have a look as well.

**Comment by tomchadwin** [comment]
Is your `_inputs` he best approach, or can an additional check be made to retrieve only `<input>`s which have a `layerId` property? I presume that your approach will perform better even than the current method because of removing `getElementsByTagName()`?

**Comment by IvanSanchez** [comment]
I think that an `this._inputs` approach should be better. There's no reason to query the DOM if we know when each `<input>` is being created and destroyed.

If removing an input is very tricky (it should be just a call to `this._inputs.splice()`), then try the alternate method of checking for the property.

**Comment by tomchadwin** [comment]
First attempt failed:

https://github.com/tomchadwin/Leaflet/commit/06a75177bc83d61b5e8111821481999a0c54f5bb

I don't think I know enough to get this done, but if I'm on the right track, great.

**Comment by serdarkacka** [comment]
I'll take a look too.

**Comment by Caramelo18** [comment]
@tomchadwin I have been trying to fix this bug and I've looked into your solution and I think it is correct. The solution I found was to change the playgroud @IvanSanchez provided.
I have made two changes and I think it worked:

- switched from "L.DomEvent" to "L.DomEvent.on" (lines 38 and 42)
- switched from "ev.value" to "ev.target.value" (lines 39 and 43)

Is this solution acceptable?

**Comment by despoinakaz** [comment]
Hello,
is this issue still open? Can I have a look at it too?

**Comment by perliedman** [comment]
@despoinakaz to my knowledge, no one has submitted a fix for this, so feel free to give it a spin.

**Comment by tomchadwin** [comment]
@despoinakaz If you can make something of what @Caramelo18 says above, that would be a start. Any thoughts, @IvanSanchez?

**Comment by despoinakaz** [comment]
@tomchadwin I made the same changes with you in Control.Layers.js and by altering the playground example,  I can now change the opacity of the layers.

**Comment by tomchadwin** [comment]
@despoinakaz I just couldn't understand the changes you said you made to the playground. Didn't know which files you altered.

**Comment by despoinakaz** [comment]
@tomchadwin I made the changes that @Caramelo18 had proposed.

**Comment by iZucken** [comment]
Hello. I assume that this issue is still open.

[Here is a commit addressing it]https://github.com/iZucken/Leaflet/commit/04aa6b5aae4c2b12db75cf660a77268b9d4bce3c)

[It works in "playground"](http://playground-leaflet.rhcloud.com/hewu/edit?html,js,output), until you want to modify existing control. After that any custom controls will fail.

I also had a difficult time figuring out a good test for it. Also, please - do notice commit's description.

**Comment by bibhuk** [comment]
I am new to open source, but have multiple years of professional experience.  Love this project, it's towards the end but would love to contribute.

**Comment by awlui** [comment]
Newbie here looking to take a crack at Open Source with this issue. Have never used leaflet before so I'll pick at the documentation before I dig in to the fix. I see that the issue is almost over a year old so don't know how issue resolution works 0.o

**Comment by MhdTlb** [comment]
The issue is open? 
Can I try to fix it?

**Comment by perliedman** [comment]
Sorry, we missed closing this. It was fixed by #5544 that has been merged, so this should work in the newly released 1.1.0 version.
